tests/helpers.bash

```diff
@@ -139,6 +139,22 @@ function wait_for_policy_enforcement {
     done
 }
 
+function wait_for_cilium_k8s_ep_gen {
+    export cilium_k8s_npods=$(kubectl get pods -n kube-system | grep cilium | wc -l)
+    set +x
+    { for i in $(seq 1 ${cilium_k8s_npods}); do
+        while true; do
+            cilium_pod_id=$(kubectl get pods -n kube-system | grep cilium | awk "NR==$i{ print \$1 }")
+            if ! kubectl exec -n kube-system -i ${cilium_pod_id} cilium endpoint list | grep regenerating; then
```

Comment:
I don't think this is safe. This assumes that by the name you call this, at least endpoint has already been marked to be in regenerating state. That might not be true, regeneration might be delayed to after the first iteration of this loop has been executed.